spiped/main.c

```diff
@@ -292,12 +321,15 @@ main(int argc, char * argv[])
 
 	/* Start accepting connections. */
 	if ((dispatch_cookie = dispatch_accept(s, opt_t, opt_R ? 0.0 : opt_r,
-	    sas_t, opt_d, opt_f, opt_g, opt_j, K, opt_n, opt_o,
-	    opt_1 ? &conndone : NULL)) == NULL) {
+	    sas_t, opt_d, opt_f, opt_g, opt_j, K, opt_n, opt_o, opt_1,
+	    &conndone)) == NULL) {
 		warnp("Failed to initialize connection acceptor");
 		goto err5;
 	}
 
+	/* Check periodically whether a signal was received. */
+	events_timer_register_double(graceful_shutdown, dispatch_cookie, 1.0);
```

Comment:
@gperciva might want to record the cookie so that he can deregister the callback in the event of a clean exit so that it isn't flagged as a memory leak.